gc3apps/irm.uzh.ch/gbeastdaemon.py

```diff
@@ -0,0 +1,119 @@
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+#
+__docformat__ = 'reStructuredText'
+__version__ = '$Revision$'
+
+
+import gc3libs
+from gc3libs.cmdline import SessionBasedDaemon
+import os
+from gc3libs.quantity import GiB, MiB
+
+class GBeastApp(gc3libs.Application):
+    def __init__(self, beast, jarfile, path, ncores, **extra):
+        self.fname = os.path.basename(path)
+        self.infile = path
+        self.extra = extra
+        extra['output_dir'] = self.fname[:-4] + '.d'
+        extra['requested_cores'] = ncores
+        extra['requested_memory'] = ncores*4000*MiB
+        extra['jobname'] = "GBeastApp.%s" % self.fname
+        args = ['java',
+                '-Xmx%dm' % extra['requested_memory'].amount(MiB),
+                '-jar', jarfile,
+                '-threads', ncores,
+                '-beagle_instances', ncores,
+                self.fname]
+
+        gc3libs.Application.__init__(
+            self,
+            arguments = args,
+            inputs = [self.infile],
+            outputs = gc3libs.ANY_OUTPUT,
+            stdout = 'stdout.txt',
+            stderr = 'stderr.txt',
+            **extra)
+
+class GBeastDaemon(SessionBasedDaemon):
+    """This script gets a directory as only argument. For each file in the
+    directory will run a GBeastApp application.
+
+    It also check, through inotify, if new files are added to the
+    directory, in that case, a new application is added to the
+    session.
+
+    """
+    version = '0.1'
+
+    def setup_options(self):
+        """Script to parallelize execution of BEAST over multiple input
+        files. Takes a directory containing XML files, parse"""
+
+        self.add_param('--beast1', default='/apps/BEASTv1.8.2/lib/beast.jar', help='Path to BEAST v1 jar file')
+        self.add_param('--beast2', default='/apps/BEASTv2.3.2/lib/beast.jar', help='Path to BEAST v2 jar file')
+        self.add_param('--cores', default=1, type=int, help="Amount of cores to use. Default: %(default)s")
+
+
+    def new_tasks(self, extra, epath=None, emask=0):
+        # Scan of the directory is performed in `before_main_loop`,
+        # here we only need to ensure the fake always-running task is
+        # created.
+
+        if not epath or not epath.endswith('.xml'):
+            gc3libs.log.info("Ignoring file %s since it doesn't end in .xml" % epath)
```

Comment:
If `epath` is `None` then this ends up printing the message:

    Ignoring file None since it doesn't end in .xml

while technically true, I find the message rather weird.  Can't we have two messages
for the two cases?